cmd/storage/postgresql/accrual_sys.go

```diff
@@ -0,0 +1,94 @@
+package storage
+
+import (
+	add "github.com/Tanya1515/gophermarket/cmd/additional"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+func (db *PostgreSQL) ProcessAccOrder(order add.Order) (err error) {
+	var userID int
+	tx, err := db.dbConn.Begin()
+	if err != nil {
+		return
+	}
+
+	if (order.Status == "PROCESSED") && (order.Accrual > 0) {
+		row := tx.QueryRow("SELECT user_id FROM orders WHERE id=$1", order.Number)
+		err = row.Scan(&userID)
+		if err != nil {
+			tx.Rollback()
+			return
+		}
+
+		_, err = tx.Exec("UPDATE orders SET accrual=$1 WHERE id=$2", order.Accrual, order.Number)
```

Comment:
```suggestion
		err = tx.QueryRow("UPDATE orders SET accrual=$1 WHERE id=$2 RETURNING user_id", order.Accrual, order.Number).Scan(&user_id)
```
И предыдущий SELECT становится не нужен.